remotesigning/validation.go

```diff
@@ -89,3 +97,161 @@ func CalculateWitnessHashPSBT(transaction string) (*string, error) {
 	result := hex.EncodeToString(hash)
 	return &result, nil
 }
+
+func validateScript(signing *SigningJob, seedBytes []byte) bool {
+	hardenedPath, _, err := SplitDerivationPath(signing.DerivationPath)
+	if err != nil {
+		log.Printf("Failed to parse derivation path (Previous Transaction): %v", err)
+		return false
+	}
+	_, remainingPathChangeAddress, err := SplitDerivationPath(signing.L1DerivationPath)
+	if err != nil {
+		log.Printf("Failed to parse derivation path (Next Transaction): %v", err)
+		return false
+	}
+
+	// Step 1: Derive Tx Script from Master Seed
+	masterSeedHex := hex.EncodeToString(seedBytes)
+	xpub, err := utils.GenHardenedXPub(masterSeedHex, hardenedPath, "mainnet")
+	if err != nil {
+		log.Printf("Failed to generate base xpub: %v", err)
+		return false
+	}
+	childPubkey, err := utils.DeriveChildPubKeyFromExistingXPub(xpub, remainingPathChangeAddress)
+	if err != nil {
+		log.Printf("Failed to derive pubkey: %v", err)
+		return false
+	}
+	generated_script, err := GenerateScriptFromChildPubkey(childPubkey)
+	if err != nil {
+		log.Printf("Failed to generate script from seed: %v", err)
+		return false
+	}
+	log.Printf("✓ Successfully generated script from seed! %x", generated_script)
+
+	// Step 2: Obtain Tx Script from Change Address (Directly from Transaction)
+	changeAddress, err := GetChangeAddressFromTransaction([]byte(*signing.Transaction))
+	if err != nil {
+		log.Printf("Failed to get change address from transaction: %v", err)
+		return false
+	}
+	expected_script, err := GenerateScriptFromChangeAddress(changeAddress)
+	if err != nil {
+		log.Printf("Failed to generate script from change address: %v", err)
+		return false
+	}
+	log.Printf("✓ Successfully generated script from change address! %x", expected_script)
```

Comment:
Doing both these functions is redundant, you can just get the script from the transaction. Right now this is doing the following: Transaction -> Script -> Address -> Script